specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/apimapis.json

```diff
@@ -1399,15 +1660,174 @@
         },
         "isCurrent": {
           "type": "boolean",
-          "description": "Indicates if API revision is current api revision."
+          "description": "Indicates if API revision is current api revision.",
+          "readOnly": true
         },
         "isOnline": {
           "type": "boolean",
-          "description": "Indicates if API revision is accessible via the gateway."
+          "description": "Indicates if API revision is accessible via the gateway.",
+          "readOnly": true
         }
       },
       "description": "API base contract details."
     },
+    "ApiRevisionCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApiRevisionContract"
+          },
+          "description": "Page values."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Next page link if any."
+        }
+      },
+      "description": "Paged Api Revision list representation."
+    },
+    "ApiRevisionContract": {
+      "properties": {
+        "apiId": {
+          "type":"string",
+          "readOnly": true,
+          "description": "Identifier of the API Revision."
+        },
+        "apiRevision": {
+          "type":"string",
+          "readOnly": true,
+          "description": "Revision number of API."
+        },
+        "createdDateTime": {
+          "type":"string",
+          "readOnly": true,
+          "format": "date-time",
+          "description": "The time the API Revision was created."
+        },
+        "updatedDateTime": {
+          "type":"string",
+          "readOnly": true,
+          "format": "date-time",
+          "description": "The time the API Revision were updated."
+        },
+        "description": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Description of the API Revision."
+        },
+        "privateUrl": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gateway URL for accessing the non-current API Revision."
+        },
+        "isOnline": {
+          "type": "boolean",
+          "readOnly": true,
+          "description": "Indicates if API revision is the current api revision."
+        },
+        "isCurrent": {
+          "type": "boolean",
+          "readOnly": true,
+          "description": "Indicates if API revision is accessible via the gateway."
+        }
+      }
+    },
+    "ApiRevisionInfoContract" : {
+        "properties": {
+          "sourceApiId": {
+            "type": "string",
+            "description": "Id of API to be used to create the revision from."
+          },
+          "ApiVersionName": {
+            "type": "string",
+            "description": "Version identifier for the new API Version."
+          },
+          "ApiVersionDescription": {
+            "type": "string",
```

Comment:
is there some min max to this.

Reply:
That field has now been removed.  It is no longer used.